### DISCUSSION TITLE
posts/Go/install/

### DISCUSSION THREAD

**Discussion by Q1mi** [Q1mi/BlogComments#365] [Announcements]
https://www.liwenzhou.com/posts/Go/install_go_dev/ 

 李文周的Blog

**Comment by cg430** [comment]
打卡

**Comment by chaofu** [comment]
学习go 开始了

**Comment by jie-521** [comment]
学习  坚持！！！

**Comment by daoofcode** [comment]
在linux安装中解压的命令那有个错误，应该是local,您误写成lcoal了

**Comment by Q1mi** [comment]
> @codebyzhp
> 在linux安装中解压的命令那有个错误，应该是local,您误写成lcoal了

感谢反馈，已更正。

**Comment by yzytmac** [comment]
所有 的工程都写在src中不对吧,环境变量直接/usr/local/go/bin这样配吧

**Comment by SmeagolYe** [comment]
学习go的第一天！坚持！

**Comment by Q1mi** [comment]
> @yzytmac
> 所有 的工程都写在src中不对吧,环境变量直接/usr/local/go/bin这样配吧

旧版本没有go module模式的时候，确实是需要把源代码都放在src中的，现在都使用go module模式就不强制源代码必须放在src下了。

**Comment by gjwlinux** [comment]
还是老司机的博客稳

**Comment by luoZhong98** [comment]
您好，我想问一下，在使用了module管理之后，go install和go build除了生成exe的位置不一样还有什么区别嘛？在用go path的时候go install -i在pkg文件夹里会生成.a文件，但是用了module之后和build一样都不生成.a了呀。

**Comment by FullStackUncle** [comment]
增加一个新手出现的问题  修复记录
安装VSCode后第一个代码会出现    package main 下有波浪线的情况，使用go build会直接报错。
主要原因是 设置 go env的环境变量中  set GO111MODULE=on   ，这个表示使用了 go mod init 
mod 模块。把这个设置成 set GO111MODULE=auto 即可。这样执行代码就不会出现波浪线，也可以
执行 go build命令了
命令：go env -w GO111MODULE=auto

**Comment by maxxxxyang** [comment]
PHP老开发前来报到 争取早日掌握

**Comment by YSH-19980902** [comment]
周周你好，
         1.是不是现在安装go只需配置下https://goproxy.cn/上的两个环境变量就行了？其他环境变量不用管。因为看你之前的视频好像必须什么src bin pkg什么的，初学者有些疑惑。
        2我安装go安装包时放在了D:/Go目录下，用户的环境变量gopath依然是长盘的%USERPROFILE%\go。这些没什么影响吧？

**Comment by Q1mi** [comment]
> @YSH-19980902
> 周周你好，
>          1.是不是现在安装go只需配置下https://goproxy.cn/上的两个环境变量就行了？其他环境变量不用管。因为看你之前的视频好像必须什么src bin pkg什么的，初学者有些疑惑。
>         2我安装go安装包时放在了D:/Go目录下，用户的环境变量gopath依然是长盘的%USERPROFILE%\go。这些没什么影响吧？

是的

**Comment by searchff** [comment]
> @Q1mi
> > @yzytmac
> > 所有 的工程都写在src中不对吧,环境变量直接/usr/local/go/bin这样配吧
> 
> 旧版本没有go module模式的时候，确实是需要把源代码都放在src中的，现在都使用go module模式就不强制源代码必须放在src下了。

不放到src下，自己写的package引用不到啊

**Comment by pro911** [comment]
注意：Go语言1.14版本之后推荐使用go modules管理"以来"    错别字  "依赖"

**Comment by HoMeTownSoCool** [comment]
打卡

**Comment by frankjj0903** [comment]
打卡
